src/Compilers/CSharp/Portable/Symbols/SymbolWithAnnotations.cs

```diff
@@ -786,34 +791,24 @@ public override TypeSymbolWithAnnotations WithModifiers(ImmutableArray<CustomMod
                 return new NonLazyType(typeSymbol, NonNullTypesContext, isAnnotated: IsAnnotated, customModifiers);
             }
 
-            protected override TypeSymbolWithAnnotations WithNonNullTypesContext(INonNullTypesContext nonNullTypesContext)
+            public override TypeSymbolWithAnnotations WithNonNullTypesContext(INonNullTypesContext nonNullTypesContext)
             {
                 return _underlying.NonNullTypesContext == nonNullTypesContext ?
                     this :
                     new LazyNullableTypeParameter(_compilation, _underlying.WithNonNullTypesContext(nonNullTypesContext));
             }
 
-            protected override TypeSymbolWithAnnotations DoUpdate(TypeSymbol typeSymbol, ImmutableArray<CustomModifier> customModifiers)
-            {
-                if (typeSymbol.IsNullableType())
-                {
-                    return TypeSymbolWithAnnotations.Create(typeSymbol, customModifiers);
-                }
-
-                return new NonLazyType(typeSymbol, NonNullTypesContext, isAnnotated: IsAnnotated, customModifiers);
-            }
-
             public override TypeSymbolWithAnnotations AsNullableReferenceType() => this;
 
             public override TypeSymbolWithAnnotations AsNotNullableReferenceType()
             {
                 if (!_underlying.TypeSymbol.IsValueType)
                 {
-                    Debug.Assert(_underlying.IsNullable == false);
+                    //Debug.Assert(_underlying.IsNullable == false);
```

Comment:
Are these causing a cycle? If so, consider adding a prototype comment so we don't forget about them

Reply:
Yes. We shouldn't call `IsNullable` (which pulls on `NonNullTypes`) during binding, which this method is.
I'll remove the commented out code.